test/es-module/test-module-get-callsite.mjs

```diff
@@ -0,0 +1,30 @@
+import { spawnPromisified } from '../common/index.mjs';
+import * as fixtures from '../common/fixtures.mjs';
+import { strictEqual, match, throws, deepStrictEqual } from 'node:assert';
+import { test } from 'node:test';
+import { mapCallSite } from 'node:module';
+import { getCallSite } from 'node:util';
+
+test('module.mapCallSite', async () => {
+  throws(() => mapCallSite('not an object'), { code: 'ERR_INVALID_ARG_TYPE' });
+  deepStrictEqual(mapCallSite([]), []);
+  throws(() => mapCallSite({}), { code: 'ERR_INVALID_ARG_TYPE' });
+
+  const callSite = getCallSite();
+  deepStrictEqual(callSite, mapCallSite(callSite));
+  deepStrictEqual(callSite[0], mapCallSite(callSite[0]));
+});
+
+
+test('module.mapCallSite should reconstruct ts callsite', async () => {
+  const result = await spawnPromisified(process.execPath, [
+    '--no-warnings',
+    '--experimental-transform-types',
+    fixtures.path('typescript/ts/test-callsite.ts'),
+  ]);
+  const output = result.stdout.toString().trim();
+  strictEqual(result.stderr, '');
+  match(output, /lineNumber: 9/);
+  match(output, /column: 25/);
+  strictEqual(result.code, 0);
+});
```

Comment:
You might want to test that it works with a minified file as well (which can have different function names).